controller/src/beerocks/master/db/db.cpp

```diff
@@ -2298,6 +2298,11 @@ int db::get_channel_scan_dwell_time_msec(const sMacAddr &mac, bool single_scan)
 bool db::is_channel_scan_pool_supported(const sMacAddr &mac,
                                         const std::unordered_set<uint8_t> &channel_pool)
 {
+    if (channel_pool.size() == 1 && *channel_pool.begin() == SCAN_ALL_CHANNELS) {
```

Comment:
I don't like this at all. If the user wants to scan all channels, then add an API to scan on all channels. APIs should be simple - this function based on its name should only check if the given pool is supported.

Reply:
See my comment below